packages/ckeditor5-engine/src/conversion/downcastdispatcher.ts

```diff
@@ -252,48 +253,53 @@ export default class DowncastDispatcher extends EmitterMixin() {
 		markers: MarkerCollection,
 		writer: DowncastWriter
 	): void {
-		const markersAtSelection = Array.from( markers.getMarkersAtPosition( selection.getFirstPosition()! ) );
-
 		const conversionApi = this._createConversionApi( writer );
+		const markersAtSelection = Array.from( markers.getMarkersAtPosition( selection.getFirstPosition()! ) );
 
 		this._addConsumablesForSelection( conversionApi.consumable, selection, markersAtSelection );
 
-		this.fire<DowncastSelectionEvent>( 'selection', { selection }, conversionApi );
+		const isConversionCorrect = this.fire<DowncastSelectionEvent>( 'selection', { selection }, conversionApi );
```

Comment:
This is very unusual in conversion. Can't we check the selection here and simply not fire events for roots that do not have a view?

Reply:
Final decision is that we will implement this mechanism straight in `convertSelection()` but we need to add a new event for selection cleanup.